ner_v1/detectors/base_detector.py

```diff
@@ -129,6 +142,73 @@ def detect_bulk(self, messages=None, **kwargs):
                                        detection_method_list=detection_method_list,
                                        detection_language=self._target_language_script)
 
+    def _add_verification_source(self, values, verification_source_dict):
+        """
+        Add the verification source for the detected entities
+        Args:
+            values (list): List of detected text type entities
+            verification_source_dict (dict): Dict consisting of the verification source and value.
+        Returns:
+            text_entity_verified_values (list): List of dicts consisting of the key and values for the keys
+            value and verification source
+        Example:
+            values = [u'Chennai', u'New Delhi', u'chennai']
+            verification_source_dict = {"datastore_verified": True}
+
+            >> add_verification_source(values, verification_source_dict)
+                [{'datastore_verified': True, 'value': u'Chennai'},
+                 {'datastore_verified': True, 'value': u'New Delhi'},
+                 {'datastore_verified': True, 'value': u'chennai'}]
+        """
+        text_entity_verified_values = []
+        for text_entity_value in values:
+            text_entity_dict = {ENTITY_VALUE_DICT_KEY: text_entity_value}
+            text_entity_dict.update(verification_source_dict)
+            text_entity_verified_values.append(text_entity_dict)
+        return text_entity_verified_values
+
+    def combine_results(self, values, original_texts, crf_original_texts):
```

Comment:
rename `crf_original_texts ` to `free_text_entity_original_text`  and use same naming conventions wherever you are using `free_text_entity`